plugins/org.locationtech.udig.ui/src/org/locationtech/udig/internal/ui/UDIGStartup.java

```diff
@@ -22,62 +25,41 @@
 import org.eclipse.swt.widgets.Display;
 import org.eclipse.ui.IStartup;
 import org.eclipse.ui.PlatformUI;
-import org.eclipse.ui.intro.IIntroManager;
-import org.eclipse.ui.intro.IIntroPart;
-import org.geotools.referencing.ReferencingFactoryFinder;
-
 
 /**
  * This class checks that the UDIG plug-ins are able to function.
  * <p>
  * Explicitly this class checks the following:
  * <ul>
- * <li>checkForJAI() - the uDig plug-ins make use of Java Advanced Imaging, this
- * check makes sure that it is in fact available.
- * <li>checkForGDI() - in the windows platform GDI libraries must be available
- * for advanced widgets to be used (so we can have some matrix math). 
+ * <li>checkForJAI() - the uDig plug-ins make use of Java Advanced Imaging, this check makes sure
+ * that it is in fact available.
+ * <li>checkForGDI() - in the windows platform GDI libraries must be available for advanced widgets
+ * to be used (so we can have some matrix math).
  * </ul>
- * Over time it would be nice to check the HTTP proxy settings, and a few other things
- * that make it easier to work with uDig (but for now this will do).
- *  
+ * Over time it would be nice to check the HTTP proxy settings, and a few other things that make it
+ * easier to work with uDig (but for now this will do).
+ *
  * @author hbullen
  */
 public class UDIGStartup implements IStartup {
 
-	/**
-	 * The following checks are performed when the workbench is first opened.
+    /**
+     * The following checks are performed when the workbench is first opened.
      * <ul>
      * <li>checkForJAI(): optional - Dig will work with reduced functionality if JAI is not
      * available
      * <li>checkForGDI(): required - uDig will not function on WIN_32 if GDI is not present
      * </ul>
      * This method also loads some commonly used objects; subclasses may override this method (say
      * to ask the user to login)
-	 * @see org.eclipse.ui.IStartup#earlyStartup()
-	 */
-	public void earlyStartup() {
+     *
+     * @see org.eclipse.ui.IStartup#earlyStartup()
+     */
+    @Override
+    public void earlyStartup() {
         checkForGDI();
         loadCommonlyUsedObject();
-		PlatformUI.getWorkbench().addWorkbenchListener(ShutdownTaskList.instance());		
-		standByWelcome(); // uDig wants the welcome page put to the side on startup
-	}
-	/**
-	 * Make sure the Welcome screen is put into standby mode; prevents
-	 * confusion when Add Layer is used and the map is displayed behind the welcome
-	 * page (also looks better).
```

Comment:
Wondering if there is a way to decide if a layer has been added. Wait .. please have a look at IStartup Implementation of project.ui plugin (StartupOpenMaps).

_If the DND extension point will be called so that it will act as if the url/string whatever was
dropped on the application. So if a url to shapefile is on the commandline a new map will be
openned and will display the shapefile._

I guess we should consider to move implementation code from udig.ui to project.ui. Or provide the functionality still from udig.ui plugin but used from project.ui plugin. This may also be configurable, e.g. a preference constant MINIMIZE_WELCOME_ON_DATA_URL or something like that.

Reply:
I had a look at UDIGApplication which is the only class which checks obviously for programm-Arguments (`Platform.getCommandLineArgs()`) . I expected there an option to add layer on startup but couldn't find a hint.

However, socket-plugin provides functionality to add layer resources using message-header `DisplayLayer version:1.0`.

I believe its only option ath the time of writing, I'd suggest remove `standByWelcome()`as provided in this pull request. Also because of it changes default behavior in eclipse RCP applications where the Welcome-Screen is essential to guide the first-timw-user through the application.